src/Workspaces/Remote/Core/SolutionAssetProvider.cs

```diff
@@ -64,17 +59,50 @@ public async ValueTask GetAssetsAsync(PipeWriter pipeWriter, Checksum solutionCh
             // is bounded by the total size of the serialized assets.
             var localPipe = new Pipe(RemoteHostAssetSerialization.PipeOptionsWithUnlimitedWriterBuffer);
 
-            var task1 = Task.Run(() =>
+            var task1 = Task.Run(async () =>
             {
                 try
                 {
                     var stream = localPipe.Writer.AsStream(leaveOpen: false);
                     using var writer = new ObjectWriter(stream, leaveOpen: false, cancellationToken);
-                    RemoteHostAssetSerialization.WriteData(writer, singleAsset, assetMap, serializer, scope.ReplicationContext, solutionChecksum, checksums, cancellationToken);
+
+                    // This information is not actually needed on the receiving end.  However, we still send it so that the
+                    // receiver can assert that both sides are talking about the same solution snapshot and no weird invariant
+                    // breaks have occurred.
+                    solutionChecksum.WriteTo(writer);
+
+                    // If we couldn't find the scope then that means the oop execution is persisting longer than the
+                    // actual outgoing calls to it were, and it's the process of calling back to inproc to just hydrate
+                    // the solution.  Bail out gracefully.
+                    if (scope == null)
+                    {
+                        // indicate that no information is available.
+                        writer.WriteInt32(0);
+                        return false;
```

Comment:
this logic is new, but it means that in this case, the reader doesn't blow up, but instead finishes gracefully.  Then below, when we await it, we don't die.